diff --git a/generic/vendor/pineapple/hal_perf_default.te b/generic/vendor/pineapple/hal_perf_default.te
index 9ca43c9..306da7e 100644
--- a/generic/vendor/pineapple/hal_perf_default.te
+++ b/generic/vendor/pineapple/hal_perf_default.te
@@ -25,6 +25,8 @@
 # OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
 # IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+type vendor_hal_perf_default, domain, mlstrustedsubject;
+
 allow vendor_hal_perf_default self:capability kill;
 allow vendor_hal_perf_default {appdomain}:process sigkill;
 hal_client_domain(vendor_hal_perf_default, vendor_hal_poweroptservice)
diff --git a/qva/vendor/common/hal_perf_default.te b/qva/vendor/common/hal_perf_default.te
index f783d0e..cd86599 100644
--- a/qva/vendor/common/hal_perf_default.te
+++ b/qva/vendor/common/hal_perf_default.te
@@ -25,7 +25,6 @@
 # OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
 # IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-type vendor_hal_perf_default, domain, mlstrustedsubject;
 hal_server_domain(vendor_hal_perf_default, vendor_hal_perf)
 
 type vendor_hal_perf_default_exec, exec_type, vendor_file_type, file_type;
