diff --git a/BoardConfigQcom.mk b/BoardConfigQcom.mk
index ddd8857..d9a4c57 100644
--- a/BoardConfigQcom.mk
+++ b/BoardConfigQcom.mk
@@ -20,7 +20,8 @@ UM_PLATFORMS := \
     $(UM_4_19_FAMILY) \
     $(UM_5_4_FAMILY) \
     $(UM_5_10_FAMILY) \
-    $(UM_5_15_FAMILY)
+    $(UM_5_15_FAMILY) \
+    $(UM_6_1_FAMILY)
 
 LEGACY_UM_PLATFORMS := \
     $(UM_3_18_FAMILY) \
@@ -38,7 +39,8 @@ QSSI_SUPPORTED_PLATFORMS := \
     $(UM_4_19_FAMILY) \
     $(UM_5_4_FAMILY) \
     $(UM_5_10_FAMILY) \
-    $(UM_5_15_FAMILY)
+    $(UM_5_15_FAMILY) \
+    $(UM_6_1_FAMILY)
 
 BOARD_USES_ADRENO := true
 
@@ -65,6 +67,10 @@ SOONG_CONFIG_qtidisplay += \
     var1 \
     var2 \
     var3 \
+    llvmcov \
+    composer_version \
+    smmu_proxy \
+    ubwcp_headers \
     wide_color \
     target_no_raw10_custom_format \
     target_uses_aligned_ycbcr_height \
@@ -89,6 +95,10 @@ SOONG_CONFIG_qtidisplay_shift_vertical ?= 0
 SOONG_CONFIG_qtidisplay_var1 ?= false
 SOONG_CONFIG_qtidisplay_var2 ?= false
 SOONG_CONFIG_qtidisplay_var3 ?= false
+SOONG_CONFIG_qtidisplay_llvmcov ?= false
+SOONG_CONFIG_qtidisplay_smmu_proxy ?= false
+SOONG_CONFIG_qtidisplay_ubwcp_headers ?= false
+SOONG_CONFIG_qtidisplay_composer_version ?= v2
 SOONG_CONFIG_qtidisplay_wide_color ?= false
 SOONG_CONFIG_qtidisplay_target_no_raw10_custom_format ?= false
 SOONG_CONFIG_qtidisplay_target_uses_aligned_ycbcr_height ?= false
@@ -181,13 +191,13 @@ ifneq ($(filter msm8937 msm8953 msm8996,$(TARGET_BOARD_PLATFORM)),)
 endif
 
 # Enable DRM PP driver on UM platforms that support it
-ifneq ($(filter $(UM_4_9_FAMILY) $(UM_4_14_FAMILY) $(UM_4_19_FAMILY) $(UM_5_4_FAMILY) $(UM_5_10_FAMILY) $(UM_5_15_FAMILY),$(TARGET_BOARD_PLATFORM)),)
+ifneq ($(filter $(UM_4_9_FAMILY) $(UM_4_14_FAMILY) $(UM_4_19_FAMILY) $(UM_5_4_FAMILY) $(UM_5_10_FAMILY) $(UM_5_15_FAMILY) $(UM_6_1_FAMILY),$(TARGET_BOARD_PLATFORM)),)
     SOONG_CONFIG_qtidisplay_drmpp := true
     TARGET_USES_DRM_PP := true
 endif
 
 # Enable Gralloc4 on UM platforms that support it
-ifneq ($(filter $(UM_4_14_FAMILY) $(UM_4_19_FAMILY) $(UM_5_4_FAMILY) $(UM_5_10_FAMILY) $(UM_5_15_FAMILY),$(TARGET_BOARD_PLATFORM)),)
+ifneq ($(filter $(UM_4_14_FAMILY) $(UM_4_19_FAMILY) $(UM_5_4_FAMILY) $(UM_5_10_FAMILY) $(UM_5_15_FAMILY) $(UM_6_1_FAMILY),$(TARGET_BOARD_PLATFORM)),)
     SOONG_CONFIG_qtidisplay_gralloc4 := true
 endif
 
@@ -201,6 +211,19 @@ ifeq ($(filter $(UM_PLATFORMS),$(TARGET_BOARD_PLATFORM)),)
     SOONG_CONFIG_qtidisplay_displayconfig_enabled := true
 endif
 
+# Enable smmu proxy on UM platforms that support it
+ifneq ($(filter $(UM_6_1_FAMILY),$(TARGET_BOARD_PLATFORM)),)
+    SOONG_CONFIG_qtidisplay_smmu_proxy := true
+endif
+# Enable ubwcp_headers on UM platforms that support it
+ifneq ($(filter $(UM_6_1_FAMILY),$(TARGET_BOARD_PLATFORM)),)
+    SOONG_CONFIG_qtidisplay_ubwcp_headers := true
+endif
+# Check if the target uses composer version 3 and is part of composer version on every UM platforms that support it
+ifeq ($(TARGET_USES_COMPOSER3)$(filter $(UM_PLATFORMS),$(TARGET_BOARD_PLATFORM)),true)
+    SOONG_CONFIG_qtidisplay_composer_version ?= v3
+endif
+
 TARGET_ADDITIONAL_GRALLOC_10_USAGE_BITS ?= 0
 
 # Mark GRALLOC_USAGE_EXTERNAL_DISP as valid gralloc bit
@@ -210,7 +233,7 @@ TARGET_ADDITIONAL_GRALLOC_10_USAGE_BITS += | (1 << 13)
 TARGET_ADDITIONAL_GRALLOC_10_USAGE_BITS += | (1 << 21)
 
 # Mark GRALLOC_USAGE_PRIVATE_HEIF_VIDEO as valid gralloc bit on UM platforms that support it
-ifneq ($(filter $(UM_4_9_FAMILY) $(UM_4_14_FAMILY) $(UM_4_19_FAMILY) $(UM_5_4_FAMILY) $(UM_5_10_FAMILY) $(UM_5_15_FAMILY),$(TARGET_BOARD_PLATFORM)),)
+ifneq ($(filter $(UM_4_9_FAMILY) $(UM_4_14_FAMILY) $(UM_4_19_FAMILY) $(UM_5_4_FAMILY) $(UM_5_10_FAMILY) $(UM_5_15_FAMILY) $(UM_6_1_FAMILY),$(TARGET_BOARD_PLATFORM)),)
     TARGET_ADDITIONAL_GRALLOC_10_USAGE_BITS += | (1 << 27)
 endif
 
@@ -221,12 +244,12 @@ ifneq ($(filter $(MASTER_SIDE_CP_TARGET_LIST),$(TARGET_BOARD_PLATFORM)),)
 endif
 
 # Opt-in for old rmnet_data driver
-ifeq ($(filter $(UM_5_15_FAMILY),$(TARGET_BOARD_PLATFORM)),)
+ifeq ($(filter $(UM_5_15_FAMILY) $(UM_6_1_FAMILY),$(TARGET_BOARD_PLATFORM)),)
     SOONG_CONFIG_rmnetctl_old_rmnet_data := true
 endif
 
 # Use full QTI gralloc struct for GKI 2.0 targets
-ifneq ($(filter $(UM_5_10_FAMILY) $(UM_5_15_FAMILY),$(TARGET_BOARD_PLATFORM)),)
+ifneq ($(filter $(UM_5_10_FAMILY) $(UM_5_15_FAMILY) $(UM_6_1_FAMILY),$(TARGET_BOARD_PLATFORM)),)
     TARGET_GRALLOC_HANDLE_HAS_CUSTOM_CONTENT_MD_RESERVED_SIZE ?= true
     TARGET_GRALLOC_HANDLE_HAS_RESERVED_SIZE ?= true
 endif
@@ -258,6 +281,8 @@ else ifneq ($(filter $(UM_5_10_FAMILY),$(TARGET_BOARD_PLATFORM)),)
     QCOM_HARDWARE_VARIANT := sm8450
 else ifneq ($(filter $(UM_5_15_FAMILY),$(TARGET_BOARD_PLATFORM)),)
     QCOM_HARDWARE_VARIANT := sm8550
+else ifneq ($(filter $(UM_6_1_FAMILY),$(TARGET_BOARD_PLATFORM)),)
+    QCOM_HARDWARE_VARIANT := sm8650
 else
     MSM_VIDC_TARGET_LIST := $(TARGET_BOARD_PLATFORM)
     QCOM_HARDWARE_VARIANT := $(TARGET_BOARD_PLATFORM)
@@ -273,7 +298,7 @@ PRODUCT_SOONG_NAMESPACES += \
     vendor/qcom/opensource/commonsys/display \
     vendor/qcom/opensource/commonsys-intf/display
 
-ifeq ($(filter $(UM_5_10_FAMILY) $(UM_5_15_FAMILY),$(TARGET_BOARD_PLATFORM)),)
+ifeq ($(filter $(UM_5_10_FAMILY) $(UM_5_15_FAMILY) $(UM_6_1_FAMILY),$(TARGET_BOARD_PLATFORM)),)
 PRODUCT_SOONG_NAMESPACES += \
     vendor/qcom/opensource/display
 endif
@@ -288,6 +313,8 @@ ifneq ($(USE_DEVICE_SPECIFIC_DATA_IPA_CFG_MGR),true)
         PRODUCT_SOONG_NAMESPACES += hardware/qcom-caf/sm8450/data-ipa-cfg-mgr
     else ifneq ($(filter $(UM_5_15_FAMILY),$(TARGET_BOARD_PLATFORM)),)
         PRODUCT_SOONG_NAMESPACES += hardware/qcom-caf/sm8550/data-ipa-cfg-mgr
+    else ifneq ($(filter $(UM_6_1_FAMILY),$(TARGET_BOARD_PLATFORM)),)
+        PRODUCT_SOONG_NAMESPACES += hardware/qcom-caf/sm8650/data-ipa-cfg-mgr
     endif
 endif
 
diff --git a/os_pickup_sepolicy_vndr.mk b/os_pickup_sepolicy_vndr.mk
index 3d9c465..3ef45ec 100644
--- a/os_pickup_sepolicy_vndr.mk
+++ b/os_pickup_sepolicy_vndr.mk
@@ -6,6 +6,8 @@ else ifneq ($(filter $(UM_5_10_FAMILY),$(TARGET_BOARD_PLATFORM)),)
     include device/qcom/sepolicy_vndr/sm8450/SEPolicy.mk
 else ifneq ($(filter $(UM_5_15_FAMILY),$(TARGET_BOARD_PLATFORM)),)
     include device/qcom/sepolicy_vndr/sm8550/SEPolicy.mk
+else ifneq ($(filter $(UM_6_1_FAMILY),$(TARGET_BOARD_PLATFORM)),)
+    include device/qcom/sepolicy_vndr/sm8650/SEPolicy.mk
 endif
 
 -include device/lineage/sepolicy/qcom/sepolicy.mk
diff --git a/qcom_boards.mk b/qcom_boards.mk
index 417d41a..84dcf67 100644
--- a/qcom_boards.mk
+++ b/qcom_boards.mk
@@ -36,3 +36,6 @@ QCOM_BOARD_PLATFORMS += parrot
 # UM 5.15
 QCOM_BOARD_PLATFORMS += kalama
 QCOM_BOARD_PLATFORMS += crow
+
+# UM 6.1
+QCOM_BOARD_PLATFORMS += pineapple
diff --git a/qcom_defs.mk b/qcom_defs.mk
index dbb542f..e33a4e4 100644
--- a/qcom_defs.mk
+++ b/qcom_defs.mk
@@ -11,3 +11,4 @@ UM_4_19_FAMILY := kona lito bengal
 UM_5_4_FAMILY := lahaina holi
 UM_5_10_FAMILY := taro parrot
 UM_5_15_FAMILY := kalama crow
+UM_6_1_FAMILY := pineapple
